### PR TITLE
Use share.* types from stock2shop/api/pkg/share instead of sql.Null*

### PR DESCRIPTION
**Pull request by mozey**
@Ayano2000 Your comments as per keybase below

Okay, I see, so what i've seen is that with everything barring the client_id using the share types:

EG:

    case "mediumint", "int", "integer":
        goType, zero = Int32(ctx), "0"
        if d.Nullable {
            goType, zero = "sql.NullInt64", "sql.NullInt64{}"
        }

It seems to work.

But once I change that to:

    case "mediumint", "int", "integer":
        goType, zero = "int64", "0"
        if d.Nullable {
            goType, zero = "share.Int", "share.Int{}"
        }
the conversion changes from c.ClientID.int64 to int64(c.ClientID)
So one option is obviously to change the db, but that's out of the question.

The other that I can think of is doing this:

    case "mediumint", "int", "integer":
        goType, zero = "int64", "0"
        if d.Nullable {
            goType, zero = "int64", "0"
        }

which sets our idexed column to int64 regardless of wether it's nullable or not, this causes the generated code to work.

And if it's unlikely happen in practice it might be the quickest way for us to move forward without looking at how it generates the indexing functions.

I'm not sure what you think is best?
basically. I can set it to int64 as above or continue looking at how the db operations are being generated

### CODE REVIEW

**Comment by mozey** [comment]
For now lets use text replacement instead, see
https://github.com/stock2shop/api/commit/898b1b8cd57bdea929064754a9c323a96c10a672
and 
https://github.com/stock2shop/api/blob/898b1b8cd57bdea929064754a9c323a96c10a672/scripts/db/refresh.sh#L21
